### PR TITLE
Updates to setTimeout

### PR DESCRIPTION
**Pull request by wbamberg**
This PR makes some fairly major updates to the https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout page.

This started out as a fix for https://github.com/mdn/content/issues/4971, but when I looked at the page I thought https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout#reasons_for_delays needed updates.

I've made quite a few changes in the end, in particular:
* make the page into the normal structure for API reference pages
* remove old Firefox info, and generally tried to make it less about Firefox implementation details
* * removed the polyfill, and a section that suggests replacing the native `setTimeout`, which didn't seem like a good suggestion
* rewritten the "throttling" example, which didn't really make sense to me.

It's a bit hard to know where to stop with something like this, so I;m quite open to feedback if I've gone too far!

### CODE REVIEW

File: files/en-us/web/api/windoworworkerglobalscope/settimeout/index.html [hunk 1]
```diff
@@ -40,273 +38,42 @@ <h3 id="Parameters">Parameters</h3>
...
-
-<p>Yet another possibility is to use <a
-    href="/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind">function's
-    <code>bind</code></a>. Example:</p>
+<p>Timeouts are cancelled using
+  {{domxref("WindowOrWorkerGlobalScope.clearTimeout()","clearTimeout()")}}.</p>
 
-<pre class="brush: js">setTimeout(function(arg1){}.bind(undefined, 10), 1000);
-</pre>
+<p>To call a function repeatedly (e.g., every N milliseconds), consider using
```
Comment (sideshowbarker) [inline files/en-us/web/api/windoworworkerglobalscope/settimeout/index.html:67]:
```suggestion
<p>To call a function repeatedly (e.g., every <em>N</em> milliseconds), consider using
```

File: files/en-us/web/api/windoworworkerglobalscope/settimeout/index.html [hunk 2]
```diff
@@ -452,75 +167,95 @@ <h3 id="Reasons_for_delays">Reasons for delays longer than specified</h3>
...
-<h4 id="E_clampus_vitus">Clamping</h4>
+<h4 id="nested_timeouts">Nested timeouts</h4>
 
-<p>In modern browsers, <code>{{domxref("WindowOrworkerGlobalScope.setTimeout()","setTimeout()")}}</code>/{{domxref("WindowOrworkerGlobalScope.setInterval","setInterval()")}} calls are throttled to a minimum of once every 4 ms when successive calls are triggered due to callback nesting (where the nesting level is at least a certain depth), or after certain number of successive intervals.</p>
+<p>As specified in the <a href="https://www.whatwg.org/specs/web-apps/current-work/multipage/timers.html#timers">HTML standard</a>,
+browsers will enforce a minimum timeout of 4 milliseconds once a nested call to <code>setTimeout</code> has been scheduled 5 times.</p>
 
-<p>See {{anch("2 Timeouts throttled")}} example, above.</p>
+<p>This can be seen in the following example, in which we nest a call to <code>setTimeout</code> with a delay of <code>0</code> milliseconds,
+and log the delay each time the handler is called. The first four times the delay is approximately 0 milliseconds, and after that it is
```
Comment (sideshowbarker) [inline files/en-us/web/api/windoworworkerglobalscope/settimeout/index.html:176]:
```suggestion
and log the delay each time the handler is called. The first four times, the delay is approximately 0 milliseconds, and after that it is
```

File: files/en-us/web/api/windoworworkerglobalscope/settimeout/index.html [hunk 3]
```diff
@@ -540,32 +275,18 @@ <h4 id="Late_timeouts">Late timeouts</h4>
 
 <h4 id="Deferral_of_timeouts_during_pageload">Deferral of timeouts during pageload</h4>
 
-<p>Starting in Firefox 66, Firefox will defer firing setTimeout and setInterval timers
-  while the current tab is loading.   Firing is deferred until the MainThread is deemed
+<p>Firefox will defer firing <code>setTimeout()</code> timers
+  while the current tab is loading. Firing is deferred until the MainThread is deemed
```
Comment (sideshowbarker) [inline files/en-us/web/api/windoworworkerglobalscope/settimeout/index.html:279]:
```suggestion
  while the current tab is loading. Firing is deferred until the main thread is deemed
```

**Review by sideshowbarker — APPROVED** [review]
Extremely minor copy-edit suggestions. Otherwise this looks great to me.

**Comment by sideshowbarker** [comment]
> It's a bit hard to know where to stop with something like this

I hear you. I know I sometimes wish I could un-read/forget some article that has a lot of room for improvement — because it’s hard to avoid staying bothered by a kind of nagging impulse to go back and completely rewrite it from scratch, or something.

Anyway, I think you’ve hit the sweet spot here with dragging it into a big improvement over what you started with. It seems quite good enough that it’s best to land it and let further refinements happen on top of it later.

File: files/en-us/web/api/windoworworkerglobalscope/settimeout/index.html (hunk 3 above)
Comment (wbamberg) [reply files/en-us/web/api/windoworworkerglobalscope/settimeout/index.html:279]: Yeah I ran out of steam a bit by the time I got down here :)

**Pull request merged**